src/plopp/prep.py

```diff
@@ -27,7 +27,16 @@ def _convert_if_not_none(x, unit):
     return x
 
 
-def preprocess(obj, crop=None, name=''):
+def _check_input_data_size(obj):
+    limits = {1: 1_000_000, 2: 5000 * 5000}
+    if obj.ndim not in limits:
+        raise ValueError("plot can only handle 1d and 2d data.")
+    if prod(obj.shape) > limits[obj.ndim]:
+        raise ValueError("Attempting to plot data that exceeds reasonable limits. "
+                         "To disable this check, use `ignore_size=True`")
```

Comment:
```suggestion
        raise ValueError(f"Plotting data of size {size} may take very long or use an excessive amount of memory. "
                         "This is therefore disabled by default."
                         "To disable this check, use `ignore_size=True`")
```
or something like that?